src/Xamarin.Android.Tools.ApiXmlAdjuster/Tests/JavaApiTest.cs

```diff
@@ -24,6 +24,24 @@ public void TestToString ()
 			var kls = pkg.Types.First (t => t.FullName == "android.database.ContentObservable");
 			Assert.AreEqual ("[Class] android.database.ContentObservable", kls.ToString ());
 		}
+
+		[Test]
+		public void ParseName ()
+		{
+			var tn = JavaTypeName.Parse ("java.util.Function<java.util.Map.Entry<K, V>, ? extends U>");
+			Assert.AreEqual ("java.util.Function", tn.FullNameNonGeneric, "top failed to parse name");
+			Assert.AreEqual (2, tn.GenericArguments.Count, "top incorrect number of parsed generic arguments");
+			var ga1 = tn.GenericArguments [0];
+			Assert.AreEqual ("java.util.Map.Entry", ga1.FullNameNonGeneric, "genarg#0 name mismatch");
+			Assert.AreEqual (2, ga1.GenericArguments.Count, "genarg#0 incorrect number of parsed generic arguments");
+			Assert.AreEqual ("K", ga1.GenericArguments [0].FullNameNonGeneric, "genarg#0.1 name mismatch");
+			Assert.AreEqual ("V", ga1.GenericArguments [1].FullNameNonGeneric, "genarg#0.2 name mismatch");
+			var ga2 = tn.GenericArguments [1];
+			Assert.AreEqual ("?", ga2.FullNameNonGeneric, "genarg#1 name mismatch");
+			Assert.AreEqual (" extends ", ga2.BoundsType, "genarg#1 incorrect bounds type");
```

Comment:
Of course I thought about that, but I didn't find very reason to perform extra string stripping and cause more memory consumption.